### PR TITLE
add background job to rent service

### PR DESCRIPTION
**Pull request by Antik88**

### CODE REVIEW

File: Rent.Service/Rent.Service.Application/ConfigureServices.cs [hunk 1]
```diff
@@ -20,6 +23,15 @@ public static IServiceCollection AddApplicationServices(this IServiceCollection
             config.AddBehavior(typeof(IPipelineBehavior<,>), typeof(ValidationBehavior<,>));
         });
 
+        service.AddQuartz(options =>
+        {
+            options.UseJobFactory<MicrosoftDependencyInjectionJobFactory>();
```
Comment (Mmishaaa) [inline Rent.Service/Rent.Service.Application/ConfigureServices.cs:27]: code style

File: Rent.Service/Rent.Service.Infrastructure/Repository/RentRepository.cs [hunk 2]
```diff
@@ -24,23 +32,29 @@ public async Task DeleteAsync(Guid id)
         await context.Rents
            .Where(model => model.Id == id)
            .ExecuteDeleteAsync();
+
+        await Task.CompletedTask;
```
Comment (Mmishaaa) [inline Rent.Service/Rent.Service.Infrastructure/Repository/RentRepository.cs]: probably, it can be deleted

**Review by Mmishaaa — APPROVED** [review]

File: Rent.Service/Rent.Service.Application/BackgroundJobs/RentsBackgroundJob.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            bool statusChanged = await rentStatusChanger.IsStatusChanged(rent);
+
+            if (!statusChanged)
+                continue;
+
+            await rentNotificationPublisher.SendRentMessage(rent, 
+                MessageType.RentStatusChange, cancellationToken);
+        }
+    }
```
Comment (5Ka-me) [inline Rent.Service/Rent.Service.Application/BackgroundJobs/RentsBackgroundJob.cs]: Possible. But how can you improve performance of this method?

File: Rent.Service/Rent.Service.Application/Abstractions/ICancelRentRepository.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using Rent.Service.Application.Model;
+using Rent.Service.Application.Rents.Commands;
+using Rent.Service.Domain.Entity;
+
+namespace Rent.Service.Application.Abstractions;
+
+public interface ICancelRentRepository
```
Comment (ZaytsevGleb) [inline Rent.Service/Rent.Service.Application/Abstractions/ICancelRentRepository.cs]: ICancelRentRepository? What is the reason to segregate interfaces in this case? Do you have more than one realization for this interfaces? Or is it too big and u don't need all of those methods from the interfaces? Read about interface segregation and YAGNI(if u did it for the future)

**Review by ZaytsevGleb — APPROVED** [review]

**Pull request merged**